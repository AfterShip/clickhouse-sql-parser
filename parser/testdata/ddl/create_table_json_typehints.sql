CREATE TABLE t (
    j JSON(message String, a.b UInt64, max_dynamic_paths=0, SKIP x, SKIP REGEXP 're')
) ENGINE = MergeTree
ORDER BY tuple();


