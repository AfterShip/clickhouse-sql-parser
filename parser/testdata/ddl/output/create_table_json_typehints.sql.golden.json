[
  {
    "CreatePos": 0,
    "StatementEnd": 142,
    "OrReplace": false,
    "Name": {
      "Database": null,
      "Table": {
        "Name": "t",
        "QuoteType": 1,
        "NamePos": 13,
        "NameEnd": 14
      }
    },
    "IfNotExists": false,
    "UUID": null,
    "OnCluster": null,
    "TableSchema": {
      "SchemaPos": 15,
      "SchemaEnd": 105,
      "Columns": [
        {
          "NamePos": 22,
          "ColumnEnd": 102,
          "Name": {
            "Ident": {
              "Name": "j",
              "QuoteType": 1,
              "NamePos": 22,
              "NameEnd": 23
            },
            "DotIdent": null
          },
          "Type": {
            "Name": {
              "Name": "JSON",
              "QuoteType": 1,
              "NamePos": 24,
              "NameEnd": 28
            },
            "Options": {
              "LParen": 29,
              "RParen": 102,
              "Items": [
                {
                  "SkipPath": null,
                  "SkipRegex": null,
                  "MaxDynamicPaths": null,
                  "MaxDynamicTypes": null,
                  "Column": {
                    "Path": {
                      "Idents": [
                        {
                          "Name": "message",
                          "QuoteType": 1,
                          "NamePos": 29,
                          "NameEnd": 36
                        }
                      ]
                    },
                    "Type": {
                      "Name": {
                        "Name": "String",
                        "QuoteType": 1,
                        "NamePos": 37,
                        "NameEnd": 43
                      }
                    }
                  }
                },
                {
                  "SkipPath": null,
                  "SkipRegex": null,
                  "MaxDynamicPaths": null,
                  "MaxDynamicTypes": null,
                  "Column": {
                    "Path": {
                      "Idents": [
                        {
                          "Name": "a",
                          "QuoteType": 1,
                          "NamePos": 45,
                          "NameEnd": 46
                        },
                        {
                          "Name": "b",
                          "QuoteType": 1,
                          "NamePos": 47,
                          "NameEnd": 48
                        }
                      ]
                    },
                    "Type": {
                      "Name": {
                        "Name": "UInt64",
                        "QuoteType": 1,
                        "NamePos": 49,
                        "NameEnd": 55
                      }
                    }
                  }
                },
                {
                  "SkipPath": null,
                  "SkipRegex": null,
                  "MaxDynamicPaths": {
                    "NumPos": 75,
                    "NumEnd": 76,
                    "Literal": "0",
                    "Base": 10
                  },
                  "MaxDynamicTypes": null,
                  "Column": null
                },
                {
                  "SkipPath": {
                    "Idents": [
                      {
                        "Name": "x",
                        "QuoteType": 1,
                        "NamePos": 83,
                        "NameEnd": 84
                      }
                    ]
                  },
                  "SkipRegex": null,
                  "MaxDynamicPaths": null,
                  "MaxDynamicTypes": null,
                  "Column": null
                },
                {
                  "SkipPath": null,
                  "SkipRegex": {
                    "LiteralPos": 99,
                    "LiteralEnd": 101,
                    "Literal": "re"
                  },
                  "MaxDynamicPaths": null,
                  "MaxDynamicTypes": null,
                  "Column": null
                }
              ]
            }
          },
          "NotNull": null,
          "Nullable": null,
          "DefaultExpr": null,
          "MaterializedExpr": null,
          "AliasExpr": null,
          "Codec": null,
          "TTL": null,
          "Comment": null,
          "CompressionCodec": null
        }
      ],
      "AliasTable": null,
      "TableFunction": null
    },
    "Engine": {
      "EnginePos": 107,
      "EngineEnd": 142,
      "Name": "MergeTree",
      "Params": null,
      "PrimaryKey": null,
      "PartitionBy": null,
      "SampleBy": null,
      "TTL": null,
      "Settings": null,
      "OrderBy": {
        "OrderPos": 127,
        "ListEnd": 142,
        "Items": [
          {
            "OrderPos": 127,
            "Expr": {
              "Name": {
                "Name": "tuple",
                "QuoteType": 1,
                "NamePos": 136,
                "NameEnd": 141
              },
              "Params": {
                "LeftParenPos": 141,
                "RightParenPos": 142,
                "Items": {
                  "ListPos": 142,
                  "ListEnd": 142,
                  "HasDistinct": false,
                  "Items": []
                },
                "ColumnArgList": null
              }
            },
            "Alias": null,
            "Direction": ""
          }
        ]
      }
    },
    "SubQuery": null,
    "HasTemporary": false,
    "Comment": null
  }
]